packages/dedupe-mixin/package.json

```diff
@@ -13,8 +13,17 @@
   },
   "author": "open-wc",
   "homepage": "https://github.com/open-wc/open-wc/tree/master/packages/dedupe-mixin",
+  "main": "index.js",
   "module": "index.js",
   "type": "module",
+  "exports": {
```

Comment:
Not 100% sure if I'd call this a breaking change, as it surfaces all the files that were previously available. However, _maybe_ if should be and we should remove `"./src/dedupeMixin.js": "./src/dedupeMixin.js"`  and `"./index.js"`as it's all reexported through `"."` anyways?

Reply:
This is a breaking change, somebody could be importing without file extensions

Reply:
@Westbrook In #2496, it was mentioned that adding just the `"."` export would make it a breaking change, after which the pull request starved. That's why I added the deep exports, so it wouldn't be a breaking change.

Reply:
@thepassle I've pushed an additional commit which adds the extensionless variants to the exports as well, so extensionless deep imports will also still work.

Reply:
We shouldnt support extensionless imports, because it bloats import maps

Reply:
would you still consider it a breaking change to not add extensionless imports then? If you don't want to support extensionless imports, then people doing extensionless importing right now today are doing something unsupported. So if this change goes in with extension-only exports is that a breaking change still?

Reply:
I dont want to support extensionless entries in the package exports, because it leads to bloated import maps. The fact that we currently support it is not really by choice. Changing the behavior (read: preventing people from importing it in a certain way, that previously _was_ supported) is still a breaking change

Reply:
makes total sense. got it!

Reply:
@thepassle comparing notes on this…

Agree it’s a breaking change. Do you have any issues with a breaking change here? As long as the PR is updated to be processed as such, it would seem that doing this is the right path for the broadest number of users, right?

Reply:
Im ok with a breaking change, its just important that we treat it as such :p I think we should probably align the package exports in the same way we do for scoped elements. The only thing im unsure about is why scoped elements has a “types.js” entry, when it already has a “types” export condition. @Westbrook do you happen to know why this is the case?

Reply:
I've removed the named exports again, and turned the changeset into a major.

I also noticed another difference with scoped-elements: dedupe-mixin didn't have a `"types"` entry in the package.json, which means it wouldn't work with TypeScript unless the `"moduleResolution"` was set to either `"node16"`, `"nodeNext"` or `"bundler"`, as it would then take the newly added `"types"` in the `"exports"` into account. I've added this as well.